src/snowflake/snowpark/dataframe.py

```diff
@@ -1644,6 +1621,101 @@ def unpivot(
+            ...     (1, 'electronics', "JAN", 100),
+            ...     (2, 'clothes', "FEB", 200)
+            ... ], schema=["empid", "dept", "month", "sales"])
+            >>> df = df.exclude_columns("month")
+            >>> df.show()
+            -----------------------------------
+            |"EMPID"  |"DEPT"       |"SALES"  |
+            -----------------------------------
+            |1        |electronics  |100      |
+            |2        |clothes      |200      |
+            -----------------------------------
+            <BLANKLINE>
+
+        Args:
+            *cols: The columns to exclude as :class:`str`, :class:`Column` or a list of those.
+        """
+        if not cols:
+            raise ValueError("The input of exclude() cannot be empty")
+        exprs = parse_positional_args_to_list(*cols)
+        names = self._get_column_names_from_column_or_name_list(exprs)
+        normalized_names = {quote_name(n) for n in names}
+        exclude_plan = Exclude(normalized_names, self._plan)
+
+        if self._select_statement:
+            return self._with_plan(
+                SelectStatement(
+                    from_=SelectSnowflakePlan(
+                        exclude_plan, analyzer=self._session._analyzer
+                    ),
+                    analyzer=self._session._analyzer,
+                )
+            )
+        return self._with_plan(exclude_plan)
+
+    @df_api_usage
+    def rename_columns(self, column_map: Dict[ColumnOrName, str]) -> "DataFrame":
+        """Renames columns from the table. In a query, it is specified in the RENAME clause.
+
+        Example::
+
+            >>> df = session.create_dataframe([
+            ...     (1, 'electronics', "JAN", 100),
+            ...     (2, 'clothes', "FEB", 200)
+            ... ], schema=["empid", "dept", "month", "sales"])
+            >>> df = df.rename_columns({"sales":"earnings"})
+            >>> df.show()
+            ------------------------------------------------
+            |"EMPID"  |"DEPT"       |"MONTH"  |"EARNINGS"  |
+            ------------------------------------------------
+            |1        |electronics  |JAN      |100         |
+            |2        |clothes      |FEB      |200         |
+            ------------------------------------------------
+            <BLANKLINE>
+
+        Args:
+            column_map: A dictionary mapping columns to their new names
+        """
+        if not column_map:
+            raise ValueError("The input of rename() cannot be empty")
+
+        column_or_name_list, rename_list = zip(*column_map.items())
+        for name in rename_list:
+            if not isinstance(name, str):
+                raise SnowparkClientExceptionMessages.DF_CANNOT_RENAME_COLUMN_WITH_NON_STRING_VALUE(
+                    name
+                )
+
+        names = self._get_column_names_from_column_or_name_list(column_or_name_list)
+        rename_map = {k: v for k, v in zip(names, rename_list)}
```

Comment:
The names here need to be processed, e.g. enquoting, converting to uppercase.

Reply:
Yep, you can check `DataFrame.select` for this